### PR TITLE
Update the VSearchTypeButton to match the new homepage designs

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #1433 by @panchovm 

## Description

The main goal of this PR is to make it possible to set whether the search type button has a label in JavaScript, instead of relying on the CSS breakpoint. This is necessary because the breakpoints from which the button has a text label (in addition to the icon) are different: `lg` on the homepage and `xl` in the header.

Here is a list of all changes in this PR:
- it adds a prop for showing or hiding the text label, instead of relying on the screen width.
- ~~to stop the button from changing the width when the label text changes, it adds all of the non-selected search type labels, but makes them invisible (and aria-hidden). This way, the button width is as high as the longest search type name. It is also capped at 175px (if the text is wider, it is truncated) to make sure that the button doesn't grow larger than 260px (the width of the content switcher popover).~~ This change was reverted because it doesn't work well when the labels have very different lengths:

Screenshot 2022-12-27 at 6 14 35 AM

- updates the label of the button to "Select the content type: <search type>" to make its purpose clearer.
- removes the Search type buttons from the new mobile homepage, and instead always uses the new VSearchTypeButton inside the homepage search bar.
- adds all of the necessary props to the `activeType` in the `useSearchType` composable.
- ~~rotates the chevron when the button is pressed to show that it's open. I wanted to try this out because that's what I usually see happen to the dropdown arrow, but I can easily revert it.~~ Should it be rotated as in this PR, or should it stay the same as in Figma, @panchovm? - change reverted, see the reason in a comment below.

## Testing Instructions

Run the storybook to see the `VSearchTypeButton`.
Go to `/preferences` and turn the `new_header` flag on. Then, check the VSearchTypeButton in the homepage searchbar, and in the search header. It should look correct, and should have correct aria properties (so, check the Screen reader experience, too).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
> Rotates the chevron when the button is pressed to show that it's open. I wanted to try this out because that's what I usually see happen to the dropdown arrow, but I can easily revert it. Should it be rotated as in this PR, or should it stay the same as in Figma?

The chevron rotation is mostly for indicating areas that expand/collapse. Like Gutenberg does.



But in this case, the chevron lives in a button that opens a popover. Like Github does.



Therefore, I would keep the chevron orientation as in the mockup.

**Review by panchovm — DISMISSED** [review]
It looks excellent ✨ 

I tested it with VoiceOver and the label was read correctly. Although I noticed one style error:
* The "showLabel = true" variant needs to have right padding of `12px` instead of `8px`.

**Review by dhruvkb — APPROVED** [review]
Looks good and works very well.

**Review by zackkrida — APPROVED** [review]
Really great; lgtm.

File: src/components/VButton.vue [hunk 1]
```diff
@@ -317,7 +317,7 @@ a.button {
 }
 
 .action-menu {
-  @apply border-tx bg-tx text-dark-charcoal hover:border-dark-charcoal-20;
+  @apply border-tx bg-white text-dark-charcoal hover:border-dark-charcoal-20 group-focus-within:hover:border-tx;
```
Comment (obulat) [inline src/components/VButton.vue:320]:
```suggestion
  @apply border-tx bg-white text-dark-charcoal hover:border-dark-charcoal-20;
```

**Pull request merged**